geth/peers/peerpool_test.go

```diff
@@ -133,3 +134,22 @@ func (s *PeerPoolSimulationSuite) TearDown() {
 		p.Stop()
 	}
 }
+
+func TestMaxPeersOverflow(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	peer := &p2p.Server{
+		Config: p2p.Config{
+			PrivateKey:  key,
+			DiscoveryV5: true,
+			NoDiscovery: true,
+		},
+	}
+	require.NoError(t, peer.Start())
+	defer peer.Stop()
+	require.NotNil(t, peer.DiscV5)
+	pool := NewPeerPool(nil, DefaultFastSync, DefaultSlowSync, nil, true)
+	pool.handleAddedEvent(peer, &p2p.PeerEvent{})
+	require.Nil(t, peer.DiscV5)
+	pool.handleAddedEvent(peer, &p2p.PeerEvent{})
```

Comment:
A comment explaining why we run this after the Discovery V5 is shutdown would be nice. Or a separate test that checks if `handleAddedEvent` does not explode when run when the Discovery V5 is shut down.

Reply:
but that was exact purpose of this test

Reply:
Ah :D So a comment above `TestMaxPeersOverflow` would be nice. A scenario that you put in the PR description, for example.